### DISCUSSION TITLE
Compiled files differs between Maven and JDTLS

### DISCUSSION THREAD

**Discussion by abenz1267** [mfussenegger/nvim-jdtls#220] [Q&A]
Hi,

i'm currently writing some tooling and i'm running into the following issue and thought you might know what's happening here:

the files compiled via `mvn package` differ from the files compiled via JDTLS, I'm getting different md5 checksums. The maven compile plugin is set to Java 1.8.

Any pointers eventually?

Regards

**Comment by mfussenegger** [comment]
`mvn package` uses `javac` as compiler and jdtls uses the eclipse compiler. The compilers don't have to produce the exact same output, as long as they're semantically equivalent

**Comment by abenz1267** [comment]
Yeah, I found out you can tell the maven compiler plugin to use the eclipse compiler instead of the bundled javac one.
